go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -382,6 +419,37 @@ func (oc *Controller) createEgressFirewallRules(priority int, match, action, ext
 	return nil
 }
 
+func (oc *Controller) deleteEgressFirewallRule(name string) error {
+	// Find ACLs for a given egressFirewall
+	pACL := func(item *nbdb.ACL) bool {
+		if item.Name != nil {
+			return *item.Name == name
+		}
+		return false
+	}
+	egressFirewallACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, pACL)
+	if err != nil {
+		return fmt.Errorf("unable to list egress firewall ACLs, cannot cleanup old stale data, err: %v", err)
+	}
+
+	if len(egressFirewallACLs) == 0 {
+		return nil
+	} else if len(egressFirewallACLs) > 1 {
+		klog.Errorf("Duplicate ACL found for egress firewall: %s", name)
+	}
```

Comment:
do we need to return here with the above error ? in case retry help if it was transient error ?

Reply:
If we find there are more than one ACL with the same name, which there shouldn't be, we just delete them all. IMO delete shouldn't fail here, we should just log that there is some error where we configured duplicates.